tests/functional/test_new_resolver.py

```diff
@@ -531,12 +531,21 @@ def test_new_resolver_handles_prerelease(
     assert_installed(script, pkg=expected_version)
 
 
-def test_new_resolver_constraints(script):
+@pytest.mark.parametrize(
+    "constraints",
+    [
+        ["pkg<2.0", "constraint_only<1.0"],
+        # This also tests the pkg constraint don't get merged with the
+        # requirement prematurely. (pypa/pip#8134)
```

Comment:
Minor nit: I'm not completely sure this comment is helpful - the test doesn't actually verify what the comment says, it would only verify this if we hadn't implemented constraints and this test failed (which was the situation in #8134).

But more tests is unlikely to be a problem, and anyone needing context can always read #8134.